File: vollkommene-zahlen/vollkommene-zahlen.pl
#!/usr/bin/perl -w
use strict;

my $t0 = time;

$| = 1;
for(my $n = 4;; $n++) {
    my $ts = 0;
    for my $j (1 .. $n - 1) {
	if ($n % $j == 0) {
	    $ts += $j;
	}
    }
    if ($n % 100 == 0) {
        print "$n\r";
	my $t1 = time;
	if ($t1 - $t0 >= 60) {
	    printf "%d %d %.0f nps\n", $n, $t1 - $t0, $n/($t1 - $t0);
	    exit(0);
	}
    }
    print "$n $ts\n" if ($n == $ts);
}

File: vollkommene-zahlen/vollkommene-zahlen.c
#include <stdio.h>
#include <time.h>

int main(void) {

    time_t t0 = time(NULL);
    long n;

    setbuf(stdout, NULL);
    for(n = 4;; n++) {
	long ts = 0;
	long j;
	for (j = 1; j < n; j++) {
	    if (n % j == 0) {
		ts += j;
	    }
	}
	if (n % 100 == 0) {
	    time_t t1;
	    printf("%ld\r", n);
	    t1 = time(NULL);
	    if (difftime(t1, t0) >= 60) {
		printf("%ld %.0f %.0f nps\n",
		       n, difftime(t1, t0), n/difftime(t1, t0));
		exit(0);
	    }
	}
	if (n == ts) printf("%ld %ld\n", n, ts);
    }
}
